Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def resolve_url(url)
+    m = Utils.popen_read("curl", "-I", url).match /^Location: (.+)$/
+    m.nil? ? url : resolve_url(m.captures.first)
```

Comment:
If we add `-L` to the arguments here and get the last `Location:` it looks like that'll do the above without needing `proto-redir`.